torchinductor/lowering.py

```diff
@@ -2526,6 +2558,48 @@ def copy_(dst, src, non_blocking=False):
     return mutate_to(dst, src)
 
 
+@make_pointwise
+def floordiv(a, b):
+    return ops.floordiv(a, b)
+
+
+@make_pointwise
+def truncdiv(a, b):
+    return ops.truncdiv(a, b)
+
+
+@register_lowering(aten.div.Tensor_mode)
+def div_mode(a, b, rounding_mode=None):
+    both_integer = is_integer_type(a) and is_integer_type(b)
+    both_boolean = is_boolean_type(a) and is_boolean_type(b)
+    on_triton = is_triton(a) or is_triton(b)
+
+    # floordiv and truncdiv need special handling for integer tensors on Triton,
+    # see the discussion at https://github.com/openai/triton/issues/605
+    if rounding_mode == "floor":
+        assert not both_boolean, "floordiv operands can not be boolean at the same time"
+        return floordiv(a, b) if (both_integer and on_triton) else floor(div(a, b))
```

Comment:
wait, you still need to promote types to the higher one even if both are integer?